flexmeasures/data/schemas/tests/test_reporting.py

```diff
@@ -125,14 +92,56 @@ def setup_dummy_sensors(db, app):
         ),
     ],
 )
-def test_pandas_reporter_schema(
-    reporter_config, is_valid, db, app, setup_dummy_sensors
-):
+def test_pandas_reporter_config_schema(config, is_valid, db, app, setup_dummy_sensors):
 
     schema = PandasReporterConfigSchema()
 
     if is_valid:
-        schema.load(reporter_config)
+        schema.load(config)
+    else:
+        with pytest.raises(ValidationError):
+            schema.load(config)
+
+
+@pytest.mark.parametrize(
+    "input, is_valid",
+    [
+        (
+            {
+                "sensor": 1,
+                "input_sensors": {"sensor_1": {"sensor": 1}},
+                "start": "2023-06-06T00:00:00+02:00",
+                "end": "2023-06-06T00:00:00+02:00",
+            },
+            True,
+        ),
+        (
+            {
+                "input_sensors": {"sensor_1": {"sensor": 1}},
+            },
+            False,
+        ),
+        (
+            {
+                "sensor": 1,
+                "input_sensors": {
+                    "sensor_1": {
+                        "sensor": 1,
+                        "event_starts_after": "2023-06-07T00:00:00+02:00",
+                        "event_ends_before": "2023-06-07T00:00:00+02:00",
+                    }
+                },
+            },
```

Comment:
For me the main benefit of keeping the sensor in the `input` is that we can decide to have multiple ouput sensors. Maybe not as necessary in the reporters, but in the scheduler we would probably require more than 1 output sensor. 

In my opinion, this makes it easier for us to create common API endpoints and CLI commands for the `DataGenerators`. Not only that we can unify them but we can also avoid repeating the validation of the input.

Reply:
Well, just think about it. I am basically suggesting splitting the current input.json (which effectively functions as an io.json) into a separate input.json and output.json. If we don't do that then we have this odd looking situation where we have an input file with a variable called `sensor`, which specifies an output sensor, and I would look for other possibilities to remedy that, like suggesting to rename to io.json and to make the output sensor optional.

For the multi-output sensor case, could you sketch what an example input.json might look like? For example, a mapping from df column names to sensor IDs?

Reply:
Good point: it doesn't make sense to have output sensors in a variable called `input`. 

(Re)naming ideas:

For `config`:
- params/parameters
- static_parameters/static_params
- config/configuration
- parametrization
- attributes

For `input`:
- variables
- dynamic_parameters/dynamic_params
- attributes
- arguments

Example of the description of multiple outputs for the `PandasReporter`:

In JSON:
```javascript
{
	"outputs" : [
		{"dataframe" : "df1", "column" : "col1", "output_sensor" : 1},
		{"dataframe" : "df1", "column" : "col2", "output_sensor" : 2},
		{"dataframe" : "df2", "column" : "col3", "output_sensor" : 3},
		{"dataframe" : "df3", "output_sensor" : 4},
	]
}
```

In YAML:
```yaml
---
outputs:
- dataframe: df1
  column: col1
  output_sensor: 1
- dataframe: df1
  column: col2
  output_sensor: 2
- dataframe: df2
  column: col3
  output_sensor: 3
- dataframe: df3
  output_sensor: 4

```

The `column`can be optional, taking the first column as the default one. This can be useful in case of having just one column per dataframe.